admin/test/test_release.py

```diff
@@ -1177,6 +1184,13 @@ def get_dispatcher(self):
         # given RPMs, not that any metadata files are copied.
         self.assertEqual(expected_files, set(files_on_s3.keys()))
 
+        # The repository is build in self.packages_directory
+        # Ensure that that does not leak into the metadata.
+        packages_gz = files_on_s3[os.path.join(self.target_key, 'Packages.gz')]
+        with GzipFile(fileobj=StringIO(packages_gz), mode="r") as f:
+            packages_metadata = f.read()
+        self.assertNotIn(self.package_directory.path, packages_metadata)
+
```

Comment:
Put some explanation of why this test has multiple assertions - perhaps because the set up is particularly slow (but then https://testtools.readthedocs.org/en/latest/for-test-authors.html#delayed-assertions).